packages/cli/src/databases/repositories/execution.repository.ts

```diff
@@ -66,11 +76,35 @@ function parseFiltersToQueryBuilder(
 
 @Service()
 export class ExecutionRepository extends Repository<ExecutionEntity> {
+	deletionBatchSize = 100;
+
+	hardDeletionInterval: NodeJS.Timer | null = null;
+
 	constructor(
 		dataSource: DataSource,
 		private readonly executionDataRepository: ExecutionDataRepository,
 	) {
 		super(ExecutionEntity, dataSource.manager);
+
+		if (config.getEnv('executions.pruneData')) this.setPruningInterval();
+
+		this.setHardDeletionInterval();
+	}
+
+	setPruningInterval() {
+		setInterval(async () => this.pruneBySoftDeleting(), 1 * TIME.HOUR);
```

Comment:
These timers are going to run on every container. Shouldn't we add some checks to run these only on the main container?

Reply:
Ah this is a great point, we have something kinda similar that was merged by @flipswitchingmonkey yesterday for things that must run only on the main process.

Reply:
Thank you, please see [2682b3d](https://github.com/n8n-io/n8n/pull/7092/commits/2682b3d596dcbfa3aca8fd697f1f5af589217aef)